plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer.ts

```diff
@@ -562,15 +512,8 @@ export class SessionRecordingIngester {
             // TODO: immediately die? or just keep going?
         })
 
-        this.batchConsumer.consumer.on('disconnected', async (err) => {
```

Comment:
Removed this as it was race-y. If the consumer shutdown gracefully "stop" is called, which in turn triggers this disconnect handler which in turn calls stop.

Now it is always the top level server that catches the consumer loop error, and calls the approrpriate shutdown flow